src/main/cpp/events/sampleevent.cpp

```diff
@@ -372,16 +373,6 @@ void SampleEvent::mixBuffer( AudioBuffer* outputBuffer, int bufferPosition,
 
                     if ( ++_readPointer > maxBufPos )
                         _readPointer = _loopStartOffset;
-
-                }
-                else if ( loopStarted && bufferPointer > maxBufferPosition )
-                {
-                    // in case the Sequencers read offset exceeds the maximum and the
-                    // Sequencer is looping, read from start. internal _readPointer takes care of correct offset
-                    bufferPosition -= loopOffset;
-
-                    // decrement iterator as no write occurred in this iteration
-                    --i;
                 }
```

Comment:
Decrementing iterators in a loop is a tell tale sign of being high, simplify this and remove the branch.